website/pages/api-docs/secret/key-management/index.mdx

```diff
@@ -0,0 +1,540 @@
+`/keymgmt` path in Vault. Since it is possible to enable secrets engines at any
+location, please update your API calls accordingly.
+
+## Create Key
+
+This endpoint creates a named cryptographic key of a specified type. The parameters
+set cannot be changed after key creation.
+
+| Method | Path                 |
+| :----- | :------------------- |
+| `POST` | `/keymgmt/key/:name` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the key to create.
+  This is provided as part of the request URL.
+
+- `type` `(string: "rsa-2048")` – Specifies the type of cryptographic key to create. The
+  following key types are supported:
+  - `rsa-2048` - RSA with bit size of 2048 (asymmetric)
+  - `rsa-3072` - RSA with bit size of 3072 (asymmetric)
+  - `rsa-4096` - RSA with bit size of 4096 (asymmetric)
+
+### Sample Payload
+
+```json
+{
+  "type": "rsa-2048"
+}
+```
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/keymgmt/key/example-key
+```
+
+## Read Key
+
+This endpoint returns information about a named key. The `keys` object will hold information
+regarding each key version. Different information will be returned depending on the key type.
+For example, an asymmetric key will return its public key in a standard format for the type.
+
+| Method | Path                 |
+| :----- | :------------------- |
+| `GET`  | `/keymgmt/key/:name` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the key to read.
+  This is provided as part of the request URL.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    http://127.0.0.1:8200/v1/keymgmt/key/example-key
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "deletion_allowed": false,
+    "keys": {
+      "1": {
+        "creation_time": "2020-11-02T15:54:58.768473-08:00",
+        "public_key": "-----BEGIN PUBLIC KEY----- ... -----END PUBLIC KEY-----",
+        "type": "rsa-2048"
+      },
+      "2": {
+        "creation_time": "2020-11-04T16:58:47.591718-08:00",
+        "public_key": "-----BEGIN PUBLIC KEY----- ... -----END PUBLIC KEY-----",
+        "type": "rsa-2048"
+      }
+    },
+    "latest_version": 1,
+    "min_enabled_version": 1,
+    "name": "example-key",
+    "type": "rsa-2048"
+  }
+}
+```
+
+## List Keys
+
+This endpoint returns a list of all existing keys.
+
+| Method | Path           |
+| :----- | :------------- |
+| `LIST` | `/keymgmt/key` |
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request LIST \
+    http://127.0.0.1:8200/v1/keymgmt/key
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "keys": ["example-key"]
+  }
+}
+```
+
+## Update Key
+
+This endpoint updates a named key.
+
+| Method | Path                 |
+| :----- | :------------------- |
+| `PUT`  | `/keymgmt/key/:name` |
+
+### Parameters
```

Comment:
These currently can't be used during creation. A key only has a single version at creation time, so there isn't really a use for specifying a `min_enabled_version` at that point.

We'd like to allow specifying `deletion_allowed` at creation time eventually. Transit is similar in that it can only be specified for [key updates](https://www.vaultproject.io/api-docs/secret/transit#deletion_allowed).